analog-in/cpp/analog-in.cpp

```diff
@@ -47,28 +47,34 @@ void checkRoot(void)
 	return;
 }
 
-int main()
+void initPlatform(int& gpioPin)
 {
-	// check if running as root
-	checkRoot();
-
-	int gpioPin = 13;
-	string unknownPlatformMessage = "This sample uses the MRAA/UPM library for I/O access, "
-    		"you are running it on an unrecognized platform. "
-			"You may need to modify the MRAA/UPM initialization code to "
-			"ensure it works properly on your platform.\n\n";
-
 	// check which board we are running on
 	Platform platform = getPlatformType();
```

Comment:
Please test if the run for this works as I saw a bug reported where 'Platform' was not identified as  a type.

Reply:
Perhaps the sample you saw was not using the mraa namespace?

Reply:
Maybe, it is this https://jira01.devtools.intel.com/browse/ISS-2202. I will not be able to get to it.

Reply:
It does work on our physical board INTEL_UP2 with GROVE_PI_SHIELD with angle rotator as input pin. I've just checked. We have no other boards to check it